src/loader/loader.cr

```diff
@@ -11,6 +22,10 @@ module BakedFileSystem
                  # Reject hidden entities and directories
                  .reject { |path| File.directory?(path) || !(path =~ /(\/\..+)/).nil? }
 
+      if files.empty?
+        raise Error.new "no files found: #{root_path}"
```

Comment:
I think this should be solved on application level and not there. 
In application there could be new function `BakedFileSystem.exists?(path)` and you decide what to do.
And/or other option is that macro instead of 

```
 macro load(path, dir = __DIR__)
    ...
    @@files << BakedFileSystem::BakedFile.new(
        path:            parts[0],
        mime_type:       parts[1],
        size:            parts[2].to_i32,
        compressed_size: parts[3].to_i32,
    }
  end
```

generates NamedTuple where key is path to file. Then it should fail during compilation with something like 
```
Error in line 1: missing key 'css/main.css' for named tuple NamedTuple(path1: String, path2: String, ...)
```
This or that way I would remove it from there.

Reply:
I was just trying to prevent the error reported in #4 which happens when the loader returns an empty result.

At least I'd assume it is never intended to have an empty file system. If this shouldn't be an error, it's easy to skip the macro execution if `source` is empty. If it should be an error, it's best to raise it in the loader to easily report the full path where it looked for files.

Having `exists?` or access through symbols target different problems.

Reply:
#4 is partially solved with first 3 checks for nonexisting/wrong directory.
No files should be fixed and works as "empty" storage.

And how about add check for user to decide.

```
BakedFileSystem.load("empty_dir", empty: false)
```

and then throw compile time error.

Reply:
Btw. I don't think the NamedTuple access is feasible because the paths being accessed are not always known at compile time.

Reply:
My memory is really bad. I even tried it long time ago. https://github.com/schovi/baked_file_system/tree/named-tuple (WIP)